file.c

```diff
@@ -401,7 +401,7 @@ examineFile(char *path, URLFile *uf)
 		return;
 	    if ((fp = lessopen_stream(path))) {
 		UFclose(uf);
-		uf->stream = newFileStream(fp, (void (*)())pclose);
+		uf->stream = newFileStream(fp);
```

Comment:
I haven't carefully checked but won't this lead to trying to close a `FILE*` acquired via `popen` with `fclose`? The manpage states that file streams opened with `popen` must use `pclose` rather than `fclose`:

> The return value from popen() is a normal standard I/O stream in  all
> respects  save  that  it  must  be  closed  with pclose() rather than
> fclose(3)

Also if you can separate the `io_file_handle` commit from the adding `(void)` changes that'd be a bit easier to review.